pilgen/src/lib.rs

```diff
@@ -366,9 +371,34 @@ impl<T: FieldElement> ASMPILConverter<T> {
                             assert!(!value.contains_key(reg));
                             value.insert(reg.clone(), self.process_assignment_value(a));
                         }
-                        Input::Label(_) => {
+                        Input::Literal(_, LiteralKind::Label) => {
                             if let Expression::PolynomialReference(r) = a {
-                                instruction_literal_arg.push(r.name);
+                                instruction_literal_arg
+                                    .push(InstructionLiteralArg::LabelRef(r.name));
+                            } else {
+                                panic!();
+                            }
+                        }
+                        Input::Literal(_, LiteralKind::UnsignedConstant) => {
+                            // TODO evaluate expression
+                            if let Expression::Number(n) = a {
+                                assert!(n.is_in_lower_half(), "Number passed to unsigned parameter is negative or too large: {n}");
+                                instruction_literal_arg.push(InstructionLiteralArg::Number(n));
+                            } else {
+                                panic!();
+                            }
+                        }
+                        Input::Literal(_, LiteralKind::SignedConstant) => {
+                            // TODO evaluate expression
+                            if let Expression::Number(n) = a {
+                                instruction_literal_arg.push(InstructionLiteralArg::Number(n));
+                            } else if let Expression::UnaryOperation(UnaryOperator::Minus, expr) = a
+                            {
+                                if let Expression::Number(n) = *expr {
+                                    instruction_literal_arg.push(InstructionLiteralArg::Number(-n));
```

Comment:
What if this is -max, wouldn't it overflow?

Reply:
It's a finite field element, it's not two's complement.

Reply:
ok